File: ABC/060/C/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var sc = bufio.NewScanner(os.Stdin)

func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}

func main() {
	var n, t, ti, ti_plus1 int
	fmt.Scan(&n, &t)
	fmt.Scan(&ti)
	total := t

	for i := 0; i < n-1; i++ {
		fmt.Scan(&ti_plus1)
		total += min(ti_plus1-ti, t)
		ti = ti_plus1
	}
	fmt.Println(total)

}

File: ABC/060/C/main.py
# coding: utf-8

n, t = map(int, raw_input().split())
ti_list = map(int, raw_input().split())
ti = ti_list.pop(0)
total = t

for ti_plus1 in ti_list:
    total += min(ti_plus1-ti, t)
    ti = ti_plus1

print total
